community/io/src/main/java/org/neo4j/io/pagecache/impl/muninn/MuninnPageCursor.java

```diff
@@ -789,6 +789,74 @@ public int copyTo( int sourceOffset, PageCursor targetCursor, int targetOffset,
         return 0;
     }
 
+    @Override
+    public void shiftBytes( int sourceStart, int length, int shift )
+    {
+        int sourceEnd = sourceStart + length;
+        int targetStart = sourceStart + shift;
+        int targetEnd = sourceStart + length + shift;
+        if ( sourceStart < 0
+                | sourceEnd > filePageSize
+                | targetStart < 0
+                | targetEnd > filePageSize
+                | length < 0 )
+        {
+            outOfBounds = true;
+            return;
+        }
+
+        if ( shift < 0 )
+        {
+            unsafeShiftLeft( sourceStart, sourceEnd, length, shift );
+        }
+        else
+        {
+            unsafeShiftRight( sourceEnd, sourceStart, length, shift );
+        }
+    }
+
+    private void unsafeShiftLeft( int fromPos, int toPos, int length, int shift )
+    {
+        int longSteps = length >> 3;
+        if ( UnsafeUtil.allowUnalignedMemoryAccess && longSteps > 0 )
+        {
+            for ( int i = 0; i < longSteps; i++ )
+            {
+                long x = UnsafeUtil.getLong( pointer + fromPos );
+                UnsafeUtil.putLong( pointer + fromPos + shift, x );
+                fromPos += Long.BYTES;
+            }
+        }
+
+        while ( fromPos < toPos )
+        {
+            byte b = UnsafeUtil.getByte( pointer + fromPos );
+            UnsafeUtil.putByte( pointer + fromPos + shift, b );
+            fromPos++;
+        }
+    }
+
+    private void unsafeShiftRight( int fromPos, int toPos, int length, int shift )
+    {
+        int longSteps = length >> 3;
```

Comment:
Looks cool, but a bit unnecessary since the compiler will convert a `length / Long.BYTES` into just that anyway, won't it?